cmd/cloud_sql_proxy/cloud_sql_proxy.go

```diff
@@ -614,12 +615,23 @@ func main() {
 
 	logging.Infof("Ready for new connections")
 
+	go func() {
```

Comment:
I debated this a bit, but ended up running the notifies in separate goroutines to guard against any potential weirdness. Would gladly have a discussion on if this is a good idea or not.
The call dials a unix socket (but only if `NOTIFY_SOCKET` is set), and I'm wary that under the right set of unfortunate circumstances that could hang. But I don't know exactly what those circumstances would be (maybe if systemd itself is hanging)

Reply:
Do you have any idea what the impact on systems without systemd might be? 

Is this actually ready at this point in time? Or does it need to wait for `proxyCleint.Run()` to be called below.

Reply:
Looking at the docs, looks like this only kicks in if `$NOTIFY_SOCKET` is set (which you referred to above). So this shouldn't have any effect on systems without systemd present? Have you done any testing?

Reply:
Yes, exactly, the code bails early if the environment variable isn't set. So non-systemd users will not get any change in behaviour unless they set that themselves.

>Is this actually ready at this point in time?

Probably no, you are very correct that it should be holding until after `proxyClient.Run`, at least. In my testing the connection set up fast enough that I didn't see that the log statement above my addition was a bit premature. 

But yes, I've done testing, and apart from that embarrassing part, the happy cases work well. I'm not sure what the less happy ones would be, though. From digging a bit, it looks like even `proxyClient.Run` might not be entirely correct, since it just kicks off the connections starting?

Reply:
I think we're actually okay, because we start receiving connections with the `connSrc` is initialized, even if they aren't handled until `Run()` is called. However it's probably logically better to move this down closer to `.Run()` and add a comment, to help minimize the chance of a potentially long running bit sneaking in-between the two sections.